src/App.tsx

```diff
@@ -1,62 +1,117 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
 import 'bulma/css/bulma.css';
 import './App.scss';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+import { GoodsList } from './components/GoodsList';
+import goodsFromServer from './api/goods';
+
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((prev, curr) => {
+    switch (sortType) {
+      case (SortType.ALPHABET):
+        return prev.localeCompare(curr);
+
+      case (SortType.LENGTH):
+        return prev.length - curr.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setSortReverse] = useState(false);
+
+  const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+  const showResetButton = (sortType !== SortType.NONE || isReversed);
+
+  const HeandlerReset = () => {
```

Comment:
```suggestion
  const handleReset = () => {
```

Reply:
1 - we use uppercased names for classes, components, but not for everything
2 - function name should represent action